docs/git/branches.md

```diff
@@ -0,0 +1,169 @@
+---
+id: branches
+title: Branches
+sidebar_label: Lesson
+---
+
+import Feedback from "@theme/Feedback";
+
+So far, we've seen that we can use git to share our work, get review, and view history. This week, we're going to learn more about how we can use git to work together in groups, and to work on more than one thing at the same time.
+
+For this class, we're going to use the git CLI, but everything we're doing can also be done in GitHub Desktop. TODO: Link to cheat sheet with equivalences.
+
+In this class, we're going to work with a GitHub repository of recipes. TODO: Have the trainees make a repo in their account from a template - we're going to have them make and merge pull requests, so we don't want this in a CYF repo.
+
+## Learning objectives
+
+By the end of this class, you should be able to:
+
+- Explain why branches can be useful when using Git.
+- Create a branch, and add commits to it.
+- Switch between branches.
+- Create a pull request from a branch to the main branch.
+- TODO: Decide whether to include this: Create a pull request from a branch to another branch.
+- Work with another trainee on non-conflicting changes to a single branch.
+
+## What is a branch?
+
+We saw in the first Git lesson that Git stores the history of your changes as a list of commits, one after the other:
+
+TODO: Image of commit history
+
+Since then, you've been pressing a button in GitHub Desktop called "Commit to main" to make a new commit. But what does that "to main" mean? main is called a _branch_, which is a list of commits. It turns out, you can have more than one branch, and they can share some commits if they want:
+
+TODO: Image of two divergent branches with shared history, highlighting which is shared and which is divergent.
+
+### Why would we want to do this?
+
+There are a few reasons. The main one is: So you can work on more than one thing at the same time.
+
+Let's look at our repo of recipes. Right now, it contains one recipe, for a brownie in a mug. There are a couple of changes we want to make:
+
+1. We want to add an optional topping to put on top of the brownie (maybe ice cream, maybe you have a better idea). We'll need to add it both as an extra ingredient, and an extra step at the end.
+2. A lot of recipes start with a story of how the author found them, or some time they've really enjoyed the recipe. We want to add a story, before the "What you need" section.
```

Comment:
\beginsnark
Can we make sure it's a really unnecessarily long story that takes ages to scroll through before you get to the thing you actually wanted to read?
\endsnark